writethedocs/www
# Add option to locally run HTMLProofer

### What
This PR adds the option to locally run [HTMLProofer](https://github.com/gjtorikian/html-proofer). With the Python upgrade https://github.com/writethedocs/www/pull/2271 I ran into a problem with the html generation. This makes it easier to run the check without waiting for the Github actions to finish.

### How to test

1. Check out this branch
2. Open VSCode and (Re)open in container. Or use: `>Dev Containers: Reopen in Container`
3. Wait for the container build to finish.
4. Use `cd docs/` to navigate to the docs folder
5. Run `make html` to generate the html inside `docs/_build`
6. Use `cd ../` to navigate back to the root folder
7. Run the htmlproofer command used in https://github.com/writethedocs/www/blob/main/.github/workflows/ubuntu.yml
   - `htmlproofer --ignore-files "/404/,/2013/,/2014/,/2015/,/2016/,/2017/,/search\/index.html/" --allow-hash-href=true --enforce-https=false --ignore-missing-alt=true --disable-external=true docs/_build/html`
8. Make sure the command can be run and checks for html warnings.

### TODO

- [x] Make sure Github actions pass with new `ruby/setup-ruby` action.


----
📚 Documentation preview 📚: https://writethedocs-www--2285.org.readthedocs.build/
